src/library/scala/collection/immutable/HashMap.scala

```diff
@@ -2284,6 +2285,35 @@ private[immutable] final class HashMapBuilder[K, V] extends ReusableBuilder[(K,
       aliased
     }
 
+  def result[V1](f: V => V1): HashMap[K, V1] = {
+    val workList = mutable.Stack[MapNode[K, V]]()
```

Comment:
we are going to create lots of worklists (one per value). Can we just create one in the caller and reuse
Ideally with a Array based structure so push doesnt create an object